libs/api/domains/translations/src/lib/translations.service.ts

```diff
@@ -27,32 +41,74 @@ const errorHandler = (name: string) => {
 export class TranslationsService {
   constructor(private contentfulRepository: ContentfulRepository) {}
 
-  getTranslations = async (
-    namespaces?: string[],
-    lang?: string,
-  ): Promise<TranslationsDict | null> => {
-    const locale = locales.find((l) => l.code === localeMap[lang])
-
-    const result = await this.contentfulRepository
-      .getLocalizedEntries<any>('*', {
-        ['content_type']: 'namespace',
-        select: 'fields.strings',
-        'fields.namespace[in]': namespaces.join(','),
-      })
-      .catch(errorHandler('getNamespace'))
-
-    return result.items.reduce((acc, cur) => {
-      const strings = cur.fields.strings
-
-      return {
-        ...acc,
-        ...mergeWith(
-          {},
-          locale.fallbackCode ? strings?.[locale.fallbackCode] : {},
-          strings?.[locale.code],
-          (o, s) => (isEmpty(s) ? o : s),
-        ),
+  getNamespaceMessages = memoize(
+    async (namespace: string) => {
+      const results = await this.contentfulRepository
+        .getLocalizedEntries<NamespaceFields>('*', {
+          ['content_type']: 'namespace',
+          select: 'fields.strings',
+          'fields.namespace': namespace,
+        })
+        .catch(errorHandler('getNamespace'))
+
+      const messages = {
+        id: namespace,
+        is: {},
+        en: {},
+      } as Messages
+
+      for (const item of results.items) {
+        for (const contentfulLocale of Object.keys(item.fields.strings ?? {})) {
+          const strings = item.fields.strings
+
+          if (!strings) {
+            continue
+          }
+
+          for (const key of Object.keys(strings[contentfulLocale])) {
+            const locale = locales.find(
+              (item) => item.code === contentfulLocale,
+            )!.locale as Locale
+
+            if (contentfulLocale === DEFAULT_LOCALE) {
+              messages[locale][key] = strings?.[contentfulLocale]?.[key]
+            } else {
+              messages[locale][key] =
+                strings?.[contentfulLocale]?.[key] !== ''
+                  ? strings?.[contentfulLocale]?.[key]
+                  : strings?.[DEFAULT_LOCALE]?.[key]
```

Comment:
I think there might be a bug here. What if some key is undefined in English for some reason, then it will end up undefined for that locale instead of using the default locale. Should be fine to check if the string is "trueish", or just `localeStrings[key] || defaultStrings[key]`.

Reply:
I'm pretty sure that's not possible. When running the extract strings script, we create strings for both locales and default to an empty string for English messages. If it has not been edited inside contentful, we will get an empty string as value, if we don't have an Icelandic locale for it neither.

But I'm only "pretty sure" so I can add the test case just to be "100% pretty sure" :D

Reply:
Ahhh, makes sense. This should at least save us some grief if we add more locales later.